test/metabase/upload_test.clj

```diff
@@ -1337,13 +1337,22 @@
                               (merge (ordered-map/ordered-map
                                       upload/auto-pk-column-keyword ::upload-types/auto-incrementing-int-pk)))
            rows             [["Obi-Wan Kenobi"]]}}]
-  (let [driver driver/*driver*
-        db-id (mt/id)
-        table-name (ddl.i/format-name driver table-name)
-        schema-name (ddl.i/format-name driver schema-name)
+  (let [driver            driver/*driver*
+        db-id             (mt/id)
+        table-name        (ddl.i/format-name driver table-name)
+        schema-name       (ddl.i/format-name driver schema-name)
         schema+table-name (#'upload/table-identifier {:schema schema-name :name table-name})
-        insert-col-names (remove #{upload/auto-pk-column-keyword} (keys col->upload-type))
-        col-definitions (#'upload/column-definitions driver col->upload-type)]
+        ->normalized-col  (comp keyword (partial #'upload/normalize-column-name driver) name)
+        name->display     (->> (keys col->upload-type)
+                               (map name)
+                               (remove #{upload/auto-pk-column-name})
+                               ((juxt
+                                 (partial map (comp name ->normalized-col))
+                                 (partial #'upload/derive-display-names driver)))
+                               (apply zipmap))
```

Comment:
woah what is going on here 🤣

Reply:
I mean I parsed it almost fine, but don't you think `(into {} (for ...` would be much more readable than a wild `juxt`?

Reply:
Yeah I let this get a bit out of control, let me try simplify it. "Only test code" 😄